clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/PersistenceDAO.scala

```diff
@@ -38,13 +37,13 @@ trait PersistenceDAO extends LazyLogging {
     * Initialize the root storage directories for Clio documents.
     */
   def initialize(
-    indexes: Seq[ElasticsearchIndex[_]]
+    indexes: Seq[ElasticsearchIndex[_]],
+    version: String
   )(implicit ec: ExecutionContext): Future[Unit] = Future {
     // Make sure the rootPath can actually be used.
     val versionPath = rootPath.resolve(VersionFileName)
     try {
-      val versionFile =
-        Files.write(versionPath, ClioServerConfig.Version.value.getBytes)
+      val versionFile = Files.write(versionPath, version.getBytes)
```

Comment:
Also continuing my personal quest to isolate references to `ClioServerConfig` to the server initializer / `apply` methods in companion objects, to improve mockability during testing.

Reply:
> This doesn't seem that much better to me than making the test mock the config to provide the setting needed for a test.

I agree, but in the current setup it's not possible to do this. Since `ClioServerConfig` is an `object` it can't be subclassed. The scalamock library used to have a way of monkey-patching `object`s, but they dropped the functionality in their latest versions.

My ideal would be for all configs to be `case class`es, with each component that needs some config value defining their own `TypesafeConfig -> CaseClassConfig` method. I think that's too big of a change for this PR, but I can open a ticket for it.